source/MRViewer/MRSpaceMouseHandlerWindows.cpp

```diff
@@ -213,22 +217,61 @@ void SpaceMouseHandlerWindows::updateConnected_()
         {
             if ( buttons_[i] ) // button up
                 viewer.spaceMouseUp( mapButtons_[i] );
-        }
-        buttons_ = {};
-    }
+		}
+        std::for_each( buttons_.begin(), buttons_.end(), [](auto& v) { v = 0; } );
 
-    joystickIndex_ = newJoystickIndex_;
+        if ( newJoystickIndex == -1 && updateThread_.joinable() )
+            updateThread_.join();
+    }
 
-    if ( joystickIndex_ != -1 )
+    if ( newJoystickIndex != -1 )
     {
         int count;
-        const float* axesNew = glfwGetJoystickAxes( joystickIndex_, &count );
-        std::copy( axesNew, axesNew + 6, axes_.begin() );
+        const float* axesNew = glfwGetJoystickAxes( newJoystickIndex, &count );
+        std::copy( axesNew, axesNew + 6, axesOld_.begin() );
+
+        if ( joystickIndex_ == -1 )
+            startUpdateThread_();
     }
 
+	joystickIndex_ = newJoystickIndex;
     getViewerInstance().mouseController.setMouseScroll( joystickIndex_ == -1 );
 }
 
+void SpaceMouseHandlerWindows::startUpdateThread_()
+{
+    updateThreadActive_ = true;
+    axesDiff_ = { 0, 0, 0, 0, 0, 0 };
+	updateThread_ = std::thread( [&]
+    {
+        std::array<float, 6> axesDiff;
+        std::for_each( axesDiff.begin(), axesDiff.end(), [](float& v) { v = 0.f; } );
```

Comment:
mb initialize with zeros? and not use `for_each` (in other similar places too)

Reply:
fixed